yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala

```diff
@@ -1197,12 +1229,22 @@ private object Client extends Logging {
     // Set an env variable indicating we are running in YARN mode.
     // Note that any env variable with the SPARK_ prefix gets propagated to all (remote) processes
     System.setProperty("SPARK_YARN_MODE", "true")
+
+    val threadEnabled: Boolean = env.get("spark.launcher.internal.thread.enabled").
+      getOrElse("false").toBoolean
+
     val sparkConf = new SparkConf
+    if( threadEnabled) {
+      for ((key, value) <- env if key.startsWith("spark.")) {
```

Comment:
This could be `conf.setAll`, but that would entail not including `sys.env` values in this `env` argument. Which I think should be the case anyway.